database-plugins/src/main/java/io/cdap/plugin/db/batch/action/QueryAction.java

```diff
@@ -50,7 +52,7 @@ public QueryAction(QueryActionConfig config) {
 
   @Override
   public void run(BatchActionContext batchContext) throws Exception {
-    config.validate();
+    config.validate(batchContext.getFailureCollector());
```

Comment:
in run method we should call collector.getOrThrowException. In configurePipeline method, data pipeline app will throw in an exception if validation failure carries any failures. however, this does not apply to any other methods